Victor-Ugwueze/MyDiary
# #159703460 Chore Feedback API Validation

### What does this PR do ?
Improve on Validation of API requests

### Tasks to be completed ?

Add custom validation to check for empty spaces on entry title and body
Sanitize title and body from request, remove leading and trailing spaces before saving to database
Add status property to response object

Add custom validation to check for profile update fields,
 * First name and Last name should be 3 characters long or more
 * email should be required, and should be valid
 * Remove leading and trailing spaces from input fields before saving to database

Add custom validation to authentication
 * First name and Last name should be 3 characters long or more
 * Remove leading and trailing spaces from input fields

#### How should this be tested ?
You should have PostgreSQL installed, then clone this project, `checkout to chore-feedback-api-validation-159703460 ` branch
install dependencies and devDependencies  by running command `npm install`
then open `server/helpers/dbHelpers.js`  adjust the database connection to match that of your host databse, then run `npm start-dev` 
Start Postman, send request to `POST  /auth/sgnup/`

### Any background context you wish to provide ?
None

### What are the relevant Pivotal Tracker stories 
#159703460

### Screenshots 
None
